Networking/Extensions/DecodedExtension.swift

```diff
@@ -29,15 +29,21 @@ public extension Decoded {
     
 }
 
-public extension Decodable where Self: RawRepresentable, Self.RawValue == String {
+public extension Decodable where Self: RawRepresentable, Self.RawValue: Decodable {
     
     static func decode(_ json: JSON) -> Decoded<Self> {
         switch json {
-        case let .string(name): return .fromOptional(Self(rawValue: name))
-        default: return .failure(Argo.DecodeError.custom("Invalid enum value"))
+        case let .string(name) where name is Self.RawValue: return castValueToEnum(name)
+        case let .bool(value) where value is Self.RawValue: return castValueToEnum(value)
```

Comment:
So can raw value be bool or not? If it can, please add it to the examples in tests.